### ISSUE TITLE
Behavior changed incompatibly between 0.18.3 and 0.18.6?


### ISSUE THREAD

**Issue by time4tea** [Workiva/opentelemetry-dart#188]
Before going into this issue, I can confess I am very confused by the api, particularly around context propagation, having to cast to specific implementations to maybe get a root span and all the docs telling to use functions that are deprecated.

As an open source maintainer I know it is hard to keep docs up to date, but a simple realistic example test would be really helpful.

Previously I was doing this:

```dart
final span = tracer.startSpan("z");
```

This would give a span of parent:"" (empty) - a root.

But since 0.8.5 or 6, it returns a span with parent 00000000

I dont know how to convince it not to. I'm also unsure how to create an inner span.
```dart
final outer = tracer.startSpan("outer");
final inner = tracer.startSpan("inner")
```

doesn't work, but how to make it work?

I dont really want to use global things, especially as global tracer prover doesn't seem to allow itself to be reset (required for tests that might need to access it)

Sorry if this sounds like a complaint, it isn't! I'd really appreciate a real example.

The examples given seen to use a completed span as a parent (end is called before passing it down) so I don't quite get what that is doing.

Thank you!!

James

**Comment by dankhan-helmet** [comment]
Not quite, sure but this might be related to a [problem](https://github.com/Workiva/opentelemetry-dart/issues/189) I'm running into too

**Comment by time4tea** [comment]
@dankhan-helmet I am using honeycomb, and that's how I noticed this issue.. no more root spans.

I reverted to 0.18.3 temporarily.
